src/app/tests/TestAttributePathExpandIterator.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file defines read handler for a CHIP Interaction Data model
+ *
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app/AttributePathExpandIterator.h>
+#include <app/ClusterInfo.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/EventManagement.h>
+#include <app/InteractionModelDelegate.h>
+#include <app/util/mock/Constants.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/CHIPTLVDebug.hpp>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DLLUtil.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#include <nlunit-test.h>
+
+using namespace chip;
+
+namespace {
+
+using P = app::ConcreteAttributePath;
+
+void TestAllWildcard(nlTestSuite * apSuite, void * apContext)
+{
+    app::ClusterInfo clusInfo;
+
+    app::ConcreteAttributePath path;
+    P paths[] = {
+        P(0xFFFE, 0xFFF1'0001, 0x0000'FFFD), P(0xFFFE, 0xFFF1'0001, 0x0000'FFFC), P(0xFFFE, 0xFFF1'0002, 0x0000'FFFD),
```

Comment:
I wonder whether you could have written `{0xFFFE, 0xFFF1'0001, 0x0000'FFFD}` and the like, without needing `P` here.  Either way.